stl/inc/format

```diff
@@ -818,7 +820,7 @@ private:
 
     _Basic_format_arg_type _Active_state = _Basic_format_arg_type::_None;
     union {
-        monostate _No_state = monostate{};
+        _Monostate _No_state{};

```

Comment:
I think we shouldn't change the type of this member. The change is observable by `visit_format_args` (and C++26 member `basic_format_arg::visit`) with a visitor that exactly detects `monostate`.

Reply:
Yes, I was using a bad search tool and underestimated this problem.

Reply:
What about moving all monostate-related utilities(comparing/hash) back to `<xutility>`? They are now defined in `<variant>`, I think in current state this will be confusing to users if they happen to use monostate but find they are not comparable/hashable.

Reply:
I'm having difficulty reading the standardese. The standard specifies the inner "variant" as [exposition-only](https://eel.is/c++draft/format.arg#:~:text=variant%3Cmonostate,//%20exposition%20only), and there are few places referring to that `monostate` outside of `basic_format_arg`. However, `visit_format_args` says
>template<class Visitor, class Context>
>decltype(auto) visit_format_arg(Visitor&& vis, basic_format_arg<Context> arg);
> Effects: Equivalent to: return visit(std::forward<Visitor>(vis), arg.value);

So, this implies `Visitor` should assume the existence of `monostate`... Does the standard say anything special about the visitor to `basic_format_arg`? If not, I think this might be a defect in the standard...